tiledb/sm/cpp_api/fragment_info.h

```diff
@@ -177,6 +177,104 @@ class FragmentInfo {
     return std::make_pair(start, end);
   }
 
+  /** Returns the number of MBRs in the fragment with the given index. */
+  uint64_t mbr_num(uint32_t fid) const {
+    auto& ctx = ctx_.get();
+    uint64_t ret;
+    ctx.handle_error(tiledb_fragment_info_get_mbr_num(
+        ctx.ptr().get(), fragment_info_.get(), fid, &ret));
+    return ret;
+  }
+
+  /**
+   * Retrieves the MBR of the fragment with the given index on the given
+   * dimension index.
+   */
+  void get_mbr(uint32_t fid, uint32_t mid, uint32_t did, void* mbr) const {
+    auto& ctx = ctx_.get();
+    ctx.handle_error(tiledb_fragment_info_get_mbr_from_index(
+        ctx.ptr().get(), fragment_info_.get(), fid, mid, did, mbr));
+  }
+
+  /**
+   * Retrieves the MBR of the fragment with the given index on the given
+   * dimension name.
+   */
+  void get_mbr(
```

Comment:
Perhaps return a pair here (making this a templated function)? The C++ API has ways to throw exceptions if the templated type is incorrect.